packages/components/src/DataList/DataList.tsx

```diff
@@ -23,14 +27,15 @@ export function DataList<T extends DataListObject>({
   children,
   title,
   totalCount,
+  headerVisibility = { xs: true, sm: true, md: true, lg: true, xl: true },
 }: DataListProps<T>) {
-  const layout = getCompoundComponent<DataListLayoutProps<T>>(
+  const allLayouts = getCompoundComponents<DataListLayoutProps<T>>(
     children,
     DataListLayout,
-  )?.props.children;
+  );
 
-  const elementData = generateListItemElements(data);
   const headerData = generateHeaderElements(headers);
+  const mediaMatches = useGridLayoutMediaQueries();
```

Comment:
do we save any performance by calling this hook here and passing it in as props on the 2 components? Rather than calling this hook on each components?

Reply:
Honestly I don't know. I can take some time to test it soon

Reply:
The performance impact would be minimal and I assume the code running once and a stack allocation for a reference to an object would be faster than running code twice

Reply:
Tru! My only concern was the prop drilling part where we have this reusable hook that can be used by the components instead of passing it in as a prop. If saving a bit of a performance outweighs the impact of prop drilling (which is a totally valid thing to do) when we don't need to then I am good with this.